www/js/reducers/points.js

```diff
@@ -1,17 +1,174 @@
 
-import {ADD_POINT, RESET_POINTS} from '../actions/point-actions'
+import { findIndex, omit, values, has } from 'underscore'
+import { createObjectURL } from 'blob-util'
+import toId from 'to-id'
+import ngeohash from 'ngeohash'
 
-export function points(state = [], action) {
+import attach from '../util/attach'
+
+export const USER_ADD = 'pannier/points/USER_ADD';
+export const USER_RESCIND = 'pannier/points/DELETE_POINT';
+export const USER_UPDATE = 'pannier/points/USER_UPDATE';
+export const SYNC_RECEIVE = 'pannier/points/SYNC_RECEIVE';
+export const SYNC_DELETE = 'pannier/points/DELETE';
+export const RELOAD = 'pannier/points/RELOAD';
+
+export default function reducer(state = [], action) {
   switch(action.type) {
-    case ADD_POINT:
-      const newPoint = Object.assign({}, action.point, {imageBlob:action.imageBlob});
-      return [
-        ...state,
-        newPoint
-      ];
-    case RESET_POINTS:
+
+    case USER_ADD:
+      return [ ...state, action.point ];
+
+    case USER_UPDATE:
+      return state;
+
+    case USER_RESCIND:
+      return state;
+
+    case SYNC_RECEIVE:
+      const recIdx = findIndex(state, point => point._id === action.id);
+
+      if(recIdx === -1) { // Replicated point is new
+        return [ ...state, action.point ];
+      } else { // Replicated point is an edit
+        return Object.assign({}, state, { [recIdx]: action.point });
```

Comment:
Replaces the point in the list at the index `recIdx`, since javascript arrays are just objects where keys are the array indices.